### PR TITLE
MAINT-26773: Search uploaded file from Content Explorer

### PR DESCRIPTION
**Pull request by AzmiTouil**
Allow /platform/administrators and /platform/web-contributors members to search all files on the platform that they can access.

### CODE REVIEW

File: core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java [hunk 1]
```diff
@@ -34,7 +35,11 @@
 
   private FileSearchServiceConnector fileSearchServiceConnector;
   
-  private NodeHierarchyCreator nodeHierarchyCreator; 
+  private NodeHierarchyCreator nodeHierarchyCreator;
+
+  private final static String PLATFORM_WEB_CONTRIBUTORS_GROUP = "/platform/web-contributors";
+
+  private final static String PLATFORM_ADMINISTRATORS_GROUP = "/platform/administrators";
```
Comment (ahamdi) [inline core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java:42]: Please remove hardcoded group names, you have to use UserACL.getAdminGroups

File: core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java [hunk 2]
```diff
@@ -69,7 +74,10 @@ public Response searchRecentDocuments(@ApiParam(value = "Query string", required
     List<ElasticSearchFilter> recentFilters = new ArrayList<>();
     recentFilters.add(getRecentFilter());
     recentFilters.add(getFileTypesFilter());
-    recentFilters.add(getPathsFilter(Arrays.asList(Utils.SPACES_NODE_PATH, getUserPrivateNode().getPath())));
+    Identity currentIdentity = ConversationState.getCurrent().getIdentity();
+    if (!currentIdentity.isMemberOf(PLATFORM_ADMINISTRATORS_GROUP, "*") && !currentIdentity.isMemberOf(PLATFORM_WEB_CONTRIBUTORS_GROUP, "*")) {
```
Comment (ahamdi) [inline core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java:78]: Only users from Admin groups could find those files, not from web-contributors group

File: core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java (hunk 1 above)
Comment (AzmiTouil) [reply core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java]: Done!

File: core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java (hunk 2 above)
Comment (AzmiTouil) [reply core/search/src/main/java/org/exoplatform/services/wcm/search/FileSearchRestService.java]: Okay !

**Pull request merged**